packages/components/src/text-control/types.ts

```diff
@@ -3,7 +3,22 @@
  */
 import type { BaseControlProps } from '../base-control/types';
 
-export type TextControlProps = Pick<
+export type TextControlPropsType =
+	| 'date'
+	| 'datetime-local'
+	| 'email'
+	| 'number'
+	| 'password'
+	| 'tel'
+	| 'text'
+	| 'time'
+	| 'search'
+	| 'url';
+
+export type TextControlProps<
+	Type extends TextControlPropsType = 'text',
```

Comment:
I think we'll want to use the classic naming strategy for type arguments, and go with T and V for consistency with other instances in the package.

Reply:
updated